src/components/profile.ts

```diff
@@ -142,38 +150,142 @@ export const getUserProfileById = async (userId: string): Promise<UserProfile |
 
 // NOTE: data will always be just one customization, but we do not know which one, meaning many of
 // the loops are just to extract an unknown key-val pair
-export const editUserProfileById = async (userId: string, data: UserProfile): Promise<void> => {
+export const editUserProfile = async (member: GuildMember, data: UserProfile): Promise<void> => {
   const db = await openDB();
   // check if a user exists in the user_profile_table already
-  const res = await db.get(`SELECT COUNT(*) as found FROM user_profile_table where user_id = ?`, userId);
+  const res = await db.get(`SELECT COUNT(*) as found FROM user_profile_table where user_id = ?`, member.id);
   const user = res;
   let query;
+  let onlyCustomization; // declare on outerscope, we need to see if this is the year customization
+  let onlyDescription;
+  for (const [customization, description] of Object.entries(data)) {
+    // grab the only customization and its corresponding description
+    onlyCustomization = customization;
+    onlyDescription = description;
+  }
```

Comment:
Can we do `const [customization, description] = Object.entries(data)[0];` instead? I think it would be more clear without a for loop and without the `only` prefix.